frontend/components/student/StudentView.tsx

```diff
@@ -144,20 +144,17 @@ async function getEntireStudent(
     router
   );
 
-  // TODO temp solution until this gets fixed
-  const coaches = new Map<UUID, string>();
-  await axiosAuthenticated
-    .get<User[]>(Endpoints.USERS, { signal: signal })
-    .then((response) => {
-      (response.data as User[]).forEach((coach) => {
-        coaches.set(coach.id, coach.username);
-      });
-    })
-    .catch((err) => {
-      parseError(err, setError, signal, router);
+  for (const suggestion of statusSuggestionBaseList) {
+    await axiosAuthenticated.get(suggestion.suggester).then((response) => {
+      const statusSuggestion = {} as StatusSuggestion;
+      statusSuggestion.suggester = response.data as User;
+      statusSuggestion.status = suggestion.status;
+      statusSuggestion.motivation = suggestion.motivation;
+      newStudent.statusSuggestions.push(statusSuggestion);
     });
```

Comment:
I'd prefer doing this a bit different, since in a for loop every request has to wait until the previous one is finished, so I'd propose using a requester util to do all the requests at once.

```
  const suggesterMap = new Map<Url, User>();
  await getUrlMap<User>(
      statusSuggestionBaseList.map(
          (suggestionBase) => suggestionBase.suggester
      ) as string[],
      suggesterMap,
      signal,
      setError,
      router
  );

  for (const suggestion of statusSuggestionBaseList) {
    const statusSuggestion = {} as StatusSuggestion;
    statusSuggestion.suggester = suggesterMap.get(suggestion.suggester) as User;
    statusSuggestion.status = suggestion.status;
    statusSuggestion.motivation = suggestion.motivation;
    newStudent.statusSuggestions.push(statusSuggestion);
  }
```

Reply:
Ah, I wasn't aware of the `getUrlMap` requester util function. I have updated the PR to make use of this.